ruby/lib/ci/queue/redis/worker.rb

```diff
@@ -201,43 +201,68 @@ def try_to_reserve_lost_test
         def push(tests)
           @total = tests.size
 
-          if @master = redis.setnx(key('master-status'), 'setup')
-            puts "Worker electected as leader, pushing #{@total} tests to the queue."
-            puts
-
-            attempts = 0
-            duration = measure do
-              with_redis_timeout(5) do
-                redis.without_reconnect do
-                  redis.multi do |transaction|
-                    transaction.lpush(key('queue'), tests) unless tests.empty?
-                    transaction.set(key('total'), @total)
-                    transaction.set(key('master-status'), 'ready')
-
-                    transaction.expire(key('queue'), config.redis_ttl)
-                    transaction.expire(key('total'), config.redis_ttl)
-                    transaction.expire(key('master-status'), config.redis_ttl)
+          with_redis_timeout(5) do
+            redis.without_reconnect do
+              @master = leader_election do
+                puts "Worker electected as leader, pushing #{@total} tests to the queue."
+                puts
+
+                attempts = 0
+                duration = measure do
+                  begin
+                    redis.multi do |transaction|
+                      transaction.lpush(key('queue'), tests) unless tests.empty?
+                      transaction.set(key('total'), @total)
+                      transaction.set(key('master-status'), 'ready')
+
+                      transaction.expire(key('queue'), config.redis_ttl)
+                      transaction.expire(key('total'), config.redis_ttl)
+                      transaction.expire(key('master-status'), config.redis_ttl)
```

Comment:
The `set` command accept an `ex:` argument since Redis 2.6.12, so we can probably merge these commands with their respective `set`.

Reply:
We don't seem to require a Redis version though. Are we fine changing this anyway?

e.g. we just had a report of somebody using 4.8

https://github.com/Shopify/ci-queue/issues/215
https://github.com/Shopify/ci-queue/blob/main/ruby/ci-queue.gemspec#L34

Reply:
Also note, I'm talking about redis-server here, not the client library.